src/isar/models/mission.py

```diff
@@ -1,43 +1,66 @@
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from datetime import datetime
-from typing import List, Union
-from uuid import UUID
+from typing import Iterator, List, Union
+from uuid import UUID, uuid4
 
 from isar.config.settings import settings
 from isar.models.mission_metadata.mission_metadata import MissionMetadata
-from robot_interface.models.mission import STEPS
-from robot_interface.models.mission.step import DriveToPose, InspectionStep
+from robot_interface.models.mission import STEPS, Step, StepStatus
 
 
 @dataclass
-class Mission:
+class Task:
     steps: List[STEPS]
+    status: StepStatus = field(default=StepStatus.NotStarted, init=False)
```

Comment:
Does it makes sense that `Task.status` have the type `StepStatus`?

Reply:
I was thinking of renaming it to simply `Status` or something? They should be the same for both `Task` and `Step` I think. I didn't want to create a new `TaskStatus` enum with the same entries.

Reply:
A common status enum sounds good. I suggest the following. This should be sufficient, right?
```
class Status(str, Enum):
    NotStarted: str = "not_started"
    InProgress: str = "in_progress"
    Successful: str = "successful"
    PartlySuccessful: str = "partly_successful"
    Failed: str = "failed"
```

Reply:
There is also another enum named `Status` in ISAR which indicated the status of the state machine. I will see what makes sense as a change.

Reply:
And I prefer `partial_success` tbh. Are we removing `Scheduled` at the same time?

Reply:
Do we need scheduled? Is it not the same as not started?

Reply:
> There is also another enum named `Status` in ISAR which indicated the status of the state machine. I will see what makes sense as a change.

This could maybe just be attributes of the `StateMachine` object?

Reply:
And removed scheduled. The one place it was used I set it to `InProgress` instead as it made more sense.